app/controllers/foreman_snapshot_management/snapshots_controller.rb

```diff
@@ -67,8 +72,36 @@ def update
       end
     end
 
+    define_action_permission ['select_multiple_host', 'create_multiple_host'], :create
+    def select_multiple_host; end
+
+    def create_multiple_host
+      data = snapshot_params
+      snapshots_created = 0
+      @hosts.each do |h|
+        s = Snapshot.new(data.merge(host: h))
+        if s.create
+          snapshots_created += 1
+        else
+          error _('Error occurred while creating Snapshot for "%{name}": %{msg}') % { name: h.name, msg: s.errors.full_messages.to_sentence }
```

Comment:
There was an error with the debian partition table not being available. While this is definitely another issue, it showed, that this error will not be shown. Maybe is were better to use `create!` and handle all failures with `try`.

Reply:
I would not recommend using try there. Actually, this doesn't look too bad. `Shapshot#create` should return false if there were any errors.
Question is if the in invalid host object should prevent the snapshot from being created in the first place.

Reply:
I guess it should not. #26 
Actually it should never be invalid in the first place.
But i guess you suggest that `create` should catch the exception and return `false`?

Reply:
Either that or the save on the host object could be done with `host.save(validate: false)` to ignore validation errors. But I don't know if that'll actually save the audit record as I believe it also skips callbacks.